floodlight/models/kinematics.py

```diff
@@ -0,0 +1,329 @@
+                calculated in both dimensions.
+        """
+
+        if axis is None:
+            if difference == "central":
+                differences_xy = np.gradient(xy.xy, axis=0)
+            elif difference == "forward":
+                differences_xy = np.diff(xy.xy, axis=0, append=0)
+            else:
+                raise ValueError(
+                    f"Expected axis to be one of (None, 'x', 'y'), got {axis}."
+                )
+            distance_euclidean = np.hypot(
+                differences_xy[:, ::2],
+                differences_xy[:, 1::2],
+            )
+        elif axis == "x":
+            distance_euclidean = np.gradient(xy.x, axis=0)
+        elif axis == "y":
+            distance_euclidean = np.gradient(xy.y, axis=0)
+        else:
+            raise ValueError(
+                f"Expected axis to be one of (None, 'x', 'y'), got {axis}."
+            )
+
+        self._distance_euclidean_ = PlayerProperty(
+            property=distance_euclidean, name="distance_covered", framerate=xy.framerate
+        )
+
+    def distance_covered(self) -> PlayerProperty:
+        """Returns the frame-wise distance covered as computed by the fit method.
+
+        Returns
+        -------
+        distance_euclidean: XY
+            An XY object of shape (T, N), where T is the total number of frames and N is
+            the number of players. The columns contain the frame-wise euclidean distance
+            covered.
+        """
+        return self._distance_euclidean_
+
+    def cumulative_distance_covered(self) -> PlayerProperty:
+        """Returns the cumulative distance covered.
+
+        Returns
+        -------
+        cumulative_distance_euclidean: XY
+            An XY object of shape (T, N), where T is the total number of frames and N is
+            the number of players. The columns contain the cumulative euclidean distance
+            covered calculated by numpy.nancumsum() over axis=0.
+        """
+        dist = self._distance_euclidean_.property
+        cum_dist = np.nancumsum(dist, axis=0)
+        cumulative_distance = PlayerProperty(
+            property=cum_dist,
+            name="cumulative_distance_covered",
+            framerate=self._distance_euclidean_.framerate,
+        )
+        return cumulative_distance
+
+
+class VelocityModel(BaseModel):
+    """Computations for velocities of all players.
+
+    Upon calling the :func:`~VelocityModel.fit`-method, this model calculates the
+    framewise velocity for each player. The calculation can subsequently be queried by
+    calling the corresponding method:
+
+        - Frame-wise velocity --> :func:`~VelocityModel.velocity`
+
+    Notes
+    -----
+    The model calculates the velocity in the unit of the corresponding position data per
+    second. This may lead to distortions when used in combination with models that use
+    SI units or when the unit is percent.
```

Comment:
again, I'm unsure what you mean by SI units. The base model performs a ``is_metrical`` check, so maybe "non-metrical" could be a more precise wording?

Reply:
The idea was, that of specific measures (eg, power) are calculated in their respective SI units (F * v in kg*m^2/s^3). Therefore also a input in cm, would distort these calculations. But this may be thinking one step to far. I changed it to 'For input data with metrical units in metrical units, the output equals the input unit.' and leave the rest to the user.